src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -384,36 +389,51 @@ private int ParseMultiSpanHeader(TRequestHandler handler, ref SequenceReader<byt
             header = currentSlice.Slice(reader.Position, lineEnd);
         }
 
-        var headerSpan = header.ToSpan();
+        byte[]? array = null;
+        int length = (int)header.Length;
+        Span<byte> span = length <= 256 ? stackalloc byte[length] : array = ArrayPool<byte>.Shared.Rent(length);
```

Comment:
There are perf benefits to making the stackalloc size constant, i.e. 256, then slicing it to the right size. Looks like the slice already happens down below.
```suggestion
        Span<byte> span = length <= 256 ? stackalloc byte[256] : array = ArrayPool<byte>.Shared.Rent(length);
```